lib/fog/aliyun/models/compute/security_group.rb

```diff
@@ -4,82 +4,84 @@ module Fog
-        #   attributes[:security_group_rules]
-        # end
-
-        # # no one should be calling this because it doesn't do anything
-        # # useful but we deprecated the rules attribute and need to maintain the API
-        # def rules=(new_rules)
-        #   Fog::Logger.deprecation('#rules= is deprecated. Use the Fog::Compute::Openstack::SecurityGroupRules collection to create new rules.')
-        #   attributes[:security_group_rules] = new_rules
-        # end
-
-        # def save
-        #   requires :name, :description
-        #   data = service.create_security_group(name, description)
-        #   merge_attributes(data.body['security_group'])
-        #   true
-        # end
+        def authorize_security_group_sg_rule(group_id,direction="ingress",options={})
+          requires :id
+          if direction=="egress"
+            service.create_security_group_egress_sg_rule(id,group_id,options)
+          else
+            service.create_security_group_sg_rule(id,group_id,options)
+          end
+          true
+        end
 
-        # def destroy
-        #   requires :id
-        #   service.delete_security_group(id)
-        #   true
-        # end
+        def authorize_security_group_ip_rule(cidr_ip,direction="ingress",options={})
+          requires :id
+          nic_type=options[:nic_type]
+          unless nic_type
+            nic_type="internet"
+          end
+          if direction=="egress"
+            service.create_security_group_egress_ip_rule(id,cidr_ip,nic_type,options)
+          else
+            service.create_security_group_ip_rule(id,cidr_ip,nic_type,options)
+          end
+          true
+        end
 
-        # def create_security_group_rule(min, max, ip_protocol = "tcp", cidr = "0.0.0.0/0", group_id = nil)
-        #   Fog::Logger.deprecation('#create_security_group_rule is deprecated. Use the Fog::Compute::Openstack::SecurityGroupRules collection to create new rules.')
-        #   requires :id
-        #   service.create_security_group_rule(id, ip_protocol, min, max, cidr, group_id)
-        # end
+        def revoke_seurity_group_sg_rule(group_id,direction="ingress",options={})
+          requires :id
+          if direction=="egress"
+            service.delete_security_group_egress_sg_rule(id,group_id,options)
+          else
+            service.delete_security_group_sg_rule(id,group_id,options)
+          end
+          true
+        end
 
-        # def delete_security_group_rule(rule_id)
-        #   Fog::Logger.deprecation('#create_security_group_rule is deprecated. Use the Fog::Compute::Openstack::SecurityGroupRule objects to destroy rules.')
-        #   service.delete_security_group_rule(rule_id)
-        #   true
-        # end
+        def revoke_security_group_ip_rule(cidr_ip,direction="ingress",options={})
+          requires :id
+          nic_type=options[:nic_type]
+          unless nic_type
+            nic_type="internet"
+          end
```

Comment:
here you could use `nic_type = options.fetch(:nic_type, "internet")`. If it can't find the key it will use the second parameter as the default value.